### ISSUE TITLE
Docs: unclear if nested menu is possible only via `items` API, or also HTML templates?

### ISSUE THREAD

**Issue by lkraav** [vaadin/vaadin-context-menu#244]
https://vaadin.com/components/vaadin-context-menu/html-examples/context-menu-items-demos demonstrates nested menu `items` API.

https://vaadin.com/components/vaadin-context-menu/html-examples/context-menu-basic-demos#defining-content-with-polymer-templates does not make it clear whether a nested menu is also possible via HTML templates?

**Comment by Haprog** [comment]
Currently it doesn't seem to be possible to do similar nested menus via `<template>` as you can do with the `items` API (at least not easily or without JS). Thanks for pointing out that this should be clarified in the docs.

When doing nested menus via the `items` API it does quite many things to make sure that accessibility of the menu works well etc. I think this would be hard to support via a `<template>` based approach without doing a lot of modifications to the given elements in the template dynamically or without requiring a lot of boilerplate code in the `<template>`. If implemented, a `<template>` could be used to just configure an `items` property, but if doing so without just using the template contents as the content of the context menu as is, there's not much benefit in doing so (compared to current `items` API).

I would consider the `template` and `renderer` APIs more low level since those can be used to put anything inside the context menu, but when using those if you want to make something more complex you also have to implement more of it yourself.

The `items` API is more strict but also higher level of abstraction and smarter in a way that it provides the nested menus, checked items, disabled items, keyboard navigation and accessibility for you out of the box.

**Comment by web-padawan** [comment]
There is a line in the API docs describing this:

https://github.com/vaadin/vaadin-context-menu/blob/087204e497612747bdd44fb41f2eb918771df1de/src/vaadin-contextmenu-items-mixin.html#L63

If that is not clear, it means we should improve API docs readability, rather than duplicating same warnings in the demos.

**Comment by lkraav** [comment]
> There is a line in the API docs describing this:

I read it, but this wording doesn't say anything about "maybe nested also possible w/ non-items API?" topic.

**Comment by Haprog** [comment]
@web-padawan that line is not really relevant to the issue. The issue is that we don't say anywhere that it's not really possible to do nested menus with `<template>` (at least not in the way you can easily implement it using `items` instead).

**Comment by lkraav** [comment]
> The items API is more strict but also higher level of abstraction and smarter in a way that it provides the nested menus, checked items, disabled items, keyboard navigation and accessibility for you out of the box.

We're still sort of new to the custom element game and SEO is a concern, as we'd like to use these things also on public facing marketing sites etc.

Google's recent announcement of rendering pages with fully modernized Chromium sounds good, but we're still quite unclear whether using `items` JS API all across component range is SEO-safe(ish.. yet?), instead of rendering traditional HTML text-based template markup composition.

Basically `items` API turns our menu templates (and possibly others) into rendering pure JSON, instead of HTML. Not sure what the SEO impact of that is going to be.

**Comment by Haprog** [comment]
@lkraav Trying to get good SEO when dealing with shadow DOM can be tricky. If you only care about Google you should be pretty safe to even build everything in JS, though there might be a bigger delay to get that content indexed compared to a site that directly serves the HTML (e.g. server side rendering or just having static HTML).

If you need to support any other search engines (that don't execute JS) you probably need to use something like https://github.com/GoogleChrome/rendertron to generate static HTML to serve to the robots unless you can keep all the relevant content and links in document level light DOM. https://developers.google.com/search/docs/guides/fix-search-javascript (see part 5.).

I think semantically a context menu is not usually something you would want to have indexed by search engines anyway. Though maybe it can be desirable depending on your use case.

But let's not continue that topic here since this issue is not about SEO.
